sw/device/rom_exts/rom_ext_manifest_parser.c

```diff
@@ -51,8 +51,17 @@ bool rom_ext_get_signature(rom_ext_manifest_t params,
   return true;
 }
 
-uint32_t rom_ext_get_image_len(rom_ext_manifest_t params) {
-  return mmio_region_read32(params.base_addr, ROM_EXT_IMAGE_LENGTH_OFFSET);
+rom_ext_extents_t rom_ext_get_extents(rom_ext_manifest_t params) {
+  uintptr_t image_length =
+      mmio_region_read32(params.base_addr, ROM_EXT_IMAGE_LENGTH_OFFSET);
+
+  rom_ext_extents_t extents = {
+      .image_start = params.slot,
+      .signed_area_start = params.slot + ROM_EXT_SIGNED_AREA_START_OFFSET,
+      .image_end = params.slot + image_length,
+  };
```

Comment:
Nit: Can you return this value directly?

Reply:
I find this clearer that there is computation going on in the members.